### PR TITLE
Feature/when next upgrades

### PR DESCRIPTION
**Pull request by JR-Mitchell**
### What?

- Removes the argument parsing logic from `when_next.py`
- Adds new method arguments to `when_next.py` `entry()` method

### Why?

The argument parsing didn't work, due to space in commands delimiting new method parameters.
Therefore, this logic has been replaced with positional arguments.

### CODE REVIEW

File: src/cogs/when_next.py [hunk 1]
```diff
@@ -27,48 +27,44 @@ def __init__(self, bot):
...
-            ),
-            timeout=1,
-        )
+        if valid_days:
+            ipl = InteractionPipeline(ChoiceInteraction(*valid_days))
+            await ipl.send_and_watch(
+                context.channel,
+                discord.Embed(
+                    title="Hang-out planning",
+                    description="Uhhhh so when do you wanna hang out?",
```
Comment (fjebaker) [inline src/cogs/when_next.py]: I think this needs one more `h`

File: src/cogs/when_next.py [hunk 2]
```diff
@@ -27,48 +27,44 @@ def __init__(self, bot):
...
-            if from_date is False:
-                return
-        until_date = from_date + datetime.timedelta(weeks=2)
-        if UNTIL_KEY in arg_dict:
-            until_date = self._parse_date(context, arg_dict[UNTIL_KEY])
-            if until_date is False:
-                return
-        weekend_days = []
+        days_int = self._parse_int(context, "days", days)
+        if days_int is False:
```
Comment (fjebaker) [inline src/cogs/when_next.py:33]: I don't know if you used `is` over `==` deliberately, but despite being a syntax warning, this saves you if `days_int == 0` lol

File: src/cogs/when_next.py [hunk 3]
```diff
@@ -27,48 +27,44 @@ def __init__(self, bot):
...
-            from_date = self._parse_date(context, arg_dict[FROM_KEY])
-            if from_date is False:
-                return
-        until_date = from_date + datetime.timedelta(weeks=2)
-        if UNTIL_KEY in arg_dict:
-            until_date = self._parse_date(context, arg_dict[UNTIL_KEY])
-            if until_date is False:
-                return
-        weekend_days = []
+        days_int = self._parse_int(context, "days", days)
```
Comment (fjebaker) [inline src/cogs/when_next.py]: Needs to be awaited.

File: src/cogs/when_next.py [hunk 4]
```diff
@@ -27,48 +27,44 @@ def __init__(self, bot):
...
+        if days_int is False:
+            return
+        offset_int = self._parse_int(context, "start_offset", start_offset)
```
Comment (fjebaker) [inline src/cogs/when_next.py]: Await

File: src/cogs/when_next.py [hunk 5]
```diff
@@ -27,48 +27,44 @@ def __init__(self, bot):
...
+        if offset_int is False:
+            return
+        from_date = datetime.date.today() + datetime.timedelta(days=offset_int)
+        until_date = from_date + datetime.timedelta(days=days_int)
+        valid_days = []
         while from_date < until_date:
-            if from_date.weekday() > 4:
-                weekend_days.append(from_date.strftime(DATE_FORMAT))
+            if str(from_date.weekday()) in weekdays:
```
Comment (fjebaker) [inline src/cogs/when_next.py:43]: Excellent hack

File: src/cogs/when_next.py [hunk 6]
```diff
@@ -27,48 +27,44 @@ def __init__(self, bot):
...
+                context.channel,
+                discord.Embed(
+                    title="Hang-out planning",
+                    description="Uhhhh so when do you wanna hang out?",
+                    colour=discord.Colour.blue(),
+                ),
+                timeout=1,
+            )
+        else:
+            await context.send(f"Invalid combination of parameters led to no valid days to vote on")
```
Comment (fjebaker) [inline src/cogs/when_next.py]: Can you add a bit more logging so it's possible to work out if something blew up when putting `valid_days` together? Just an `info` or maybe even `debug` before the `for` loop that logs the `from_date` and `until_date` would be useful.

**Review by fjebaker — CHANGES_REQUESTED** [review]
Jawohl, gut so 👍

**Comment by fjebaker** [comment]
@JR-Mitchell have you had a chance to test this on a dev bot or is this a lucky dip?

**Comment by fjebaker** [comment]
I've been meaning to setup webhooks to get the server to automatically pull and relaunch when pushes to `main` are made, so might do that over the weekend. Should hopefully improve time-to-fail.

File: src/cogs/when_next.py (hunk 2 above)
Comment (JR-Mitchell) [reply src/cogs/when_next.py:33]: I think I just googled which way you should check falsiness in python and used that

File: src/cogs/when_next.py (hunk 5 above)
Comment (JR-Mitchell) [reply src/cogs/when_next.py:43]: ty

File: src/cogs/when_next.py (hunk 1 above)
Comment (JR-Mitchell) [reply src/cogs/when_next.py]: Resolved as of 1e786a9f

File: src/cogs/when_next.py (hunk 3 above)
Comment (JR-Mitchell) [reply src/cogs/when_next.py]: Resolved as of 1e786a9f

File: src/cogs/when_next.py (hunk 4 above)
Comment (JR-Mitchell) [reply src/cogs/when_next.py]: Resolved as of 1e786a9f

File: src/cogs/when_next.py (hunk 6 above)
Comment (JR-Mitchell) [reply src/cogs/when_next.py]: Logging added as of 1e786a9f

**Comment by JR-Mitchell** [comment]
> @JR-Mitchell have you had a chance to test this on a dev bot or is this a lucky dip?

@fjebaker lucky dip - I don't have an environment for testing set up

File: src/cogs/when_next.py [hunk 7]
```diff
@@ -48,7 +49,7 @@ async def entry(self, context, days: str = "31", weekdays: str = "567", start_of
                 context.channel,
                 discord.Embed(
                     title="Hang-out planning",
-                    description="Uhhhh so when do you wanna hang out?",
+                    description="Uhhhhh so when do you wanna hang out?",
```
Comment (fjebaker) [inline src/cogs/when_next.py:52]: 👍

**Review by fjebaker — APPROVED** [review]
LGTM

**Pull request merged**